packages/live-preview-sdk/src/inspectorMode/index.ts

```diff
@@ -0,0 +1,171 @@
+import { getAllTaggedElements, getTaggedInformation } from './utils';
+import { sendMessageToEditor } from '../helpers';
+import {
+  InspectorModeChangedMessage,
+  LivePreviewPostMessageMethods,
+  MessageFromEditor,
+  InteractionEventMethods,
+} from '../messages';
+import { TagAttributes } from './types';
+
+export class InspectorMode {
+  private defaultLocale: string;
+  private targetOrigin: string[];
+  private isScrolling: boolean = false;
+  private isResizing: boolean = false;
+  private scrollTimeout?: NodeJS.Timeout;
+  private resizeTimeout?: NodeJS.Timeout;
+  private hoveredElement?: HTMLElement;
+  private taggedElements: Element[] = [];
+
+  constructor({ locale, targetOrigin }: { locale: string; targetOrigin: string[] }) {
+    this.defaultLocale = locale;
+    this.targetOrigin = targetOrigin;
+
+    // TODO: we we need this?
+    this.onMouseOver = this.onMouseOver.bind(this);
+    this.onScroll = this.onScroll.bind(this);
+    this.handleElementInteraction = this.handleElementInteraction.bind(this);
+    this.sendAllElements = this.sendAllElements.bind(this);
+
+    // TODO: on resize do the something similar as onScroll
+    window.addEventListener('scroll', this.onScroll);
```

Comment:
should we provide a cleanup fn to remove all the listeners (maybe future improvement)